src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -295,16 +247,18 @@ private void WatchForFileSystemEventsThreadStart()
                 else if ((DoesPathPassNameFilter(eventPaths[i])) && ((_filterFlags & eventFlags[i]) != 0))
                 {
                     // Remove the base directory prefix (including trailing / that OS X adds)
-                    string relativePath = eventPaths[i].Remove(0, _fullDirectory.Length + 1);
+                    string relativePath = (eventPaths[i].Equals(_fullDirectory, StringComparison.OrdinalIgnoreCase) ? _fullDirectory : eventPaths[i].Remove(0, _fullDirectory.Length).TrimStart('/'));
 
                     // Check if this is a rename
-                    if ((eventFlags[i] & Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed) == Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed)
+                    if (IsFlagSet(eventFlags[i], Interop.EventStream.FSEventStreamEventFlags.kFSEventStreamEventFlagItemRenamed))
                     {
                         // Find the rename that is paired to this rename, which should be the next rename in the list
                         long pairedId = FindRenameChangePairedChange(i, eventPaths, eventFlags, eventIds);
                         if (pairedId == long.MinValue)
                         {
-                            OnError(new ErrorEventArgs(new ArgumentOutOfRangeException("pairedId")));
+                            // Getting here means we have a rename without a pair, meaning it should be a create for the 
+                            // move from unwatched folder to watcher folder scenario or a move from the watcher folder out
```

Comment:
Your most recent commit updated the comment to include moving out of the watched folder, but it didn't update the implementation accordingly... wouldn't moving out be treated as deletion, not creation?